File: C++/Basic/Complex/Complex.cpp
Complex(double real=0, double imaginary=0) {
        this->real = real;
        this->imaginary = imaginary;
}

~Complex() {}

double Complex::getReal() const {
        return this->real;
}

double Complex::getImaginary() const {
        return this->imaginary;
}

bool Complex::operator==(const Complex& other) const {
        return this->real == other.real && this->imaginary == other.imaginary;
}

bool Complex::operator!=(const Complex& other) const {
        return !(*this == other);
}

void Complex::operator+=(const Complex& other) {
        this->re += other.re;
        this->im += other.im;
}

void Complex::operator-=(const Complex& other) {
        this->re -= other.re;
        this->im -= other.im;
}

Complex Complex::operator+(const Complex& other) const {
        Complex c = *this;
        complexIncrement(&c, other);
        return c;
}

Complex Complex::operator-(const Complex& other) const {
        Complex c = *this;
        complexDecrement(&c, other);
        return c;
}

Complex Complex::operator*(const Complex& other) const {
        double tempRe = this->re * other.re - this->im * other.im;
        double tempIm = this->re * other.im + this->im * other.re;

        this->re = tempRe;
        this->im = tempIm;
}

std::ostream& operator<<(std::ostream& out, const Complex& complex) {
        out << complex.getReal() << "+" << complex.getImaginary() << 'i';
        return out;
}

File: Java/Basic/Complex/Complex.java
class Complex {
    private double real;
    private double imaginary;


    public Complex(double real, double imaginary) {
        this.real = real;
        this.imaginary = imaginary;
    }

    public Complex() {
        this.real = 0;
        this.imaginary = 0;
    }

    public ~Complex() {}

    public double getReal() {
        return real;
    }

    public double getImaginary() {
        return imaginary;
    }

    public boolean equals(Complex other) {
        return real == other.real && imaginary == other.imaginary;
    }

    public boolean notequals(Complex other) {
        return !(real == other.real && imaginary == other.imaginary);
    }

    public void increment(Complex other) {
        real += other.real;
        imaginary += other.imaginary;
    }

    public void decrement(Complex other) {
        real -= other.real;
        imaginary -= other.imaginary;
    }

    public Complex summary(Complex other) {
        Complex c = this;
        increment(&c, other);

        return c;
    }

    public Complex subtraction(Complex other) {
        Complex c = this;
        decrement(&c, other);

        return c;
    }

    public Complex multiple(Complex other) {
        double tempRe = real * other.real - imaginary * other.imaginary;
        double tempIm = real * other.imaginary + imaginary * other.real;

        reeal = tempRe;
        imaginary = tempIm;
    }

    public void show() {
        System.out.println(real + '+' + imaginary + 'i');
    }
